solidity/contracts/converter/ConverterBase.sol

```diff
@@ -227,16 +230,9 @@ abstract contract ConverterBase is ConverterVersion, IConverter, ContractRegistr
     {
         uint256 reserveCount = reserveTokens.length;
         for (uint256 i = 0; i < reserveCount; ++i) {
-            IERC20 reserveToken = reserveTokens[i];
+            IReserveToken reserveToken = reserveTokens[i];
 
-            uint256 amount;
-            if (reserveToken == NATIVE_TOKEN_ADDRESS) {
-                amount = address(this).balance;
-            } else {
-                amount = reserveToken.balanceOf(address(this));
-            }
-
-            safeTransfer(reserveToken, _newConverter, amount);
+            reserveToken.safeTransfer(_newConverter);
```

Comment:
I assume this is a shortcut to transfer the entire balance - risky IMO

Reply:
I'll remove it